pkg/cmd/cli/cmd/rsync/rsync.go

```diff
@@ -0,0 +1,218 @@
+package rsync
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	// RsyncRecommendedName is the recommended name for the rsync command
+	RsyncRecommendedName = "rsync"
+
+	rsyncLong = `
+Copy local files to a container
+
+This command will attempt to copy local files to a remote container. It will default to the
```

Comment:
I don't know how useful the details of the algorithm are.  Instead, I'd use this space to explain what it does (copy files that have changed locally to the remote system) and add instructions about the different operating systems (on Linux and Mac, install rsync.  On Windows, download cwRsync)